src/crypto/hasher.rs

```diff
@@ -1,114 +1,86 @@
-use super::blake3_sequence::Blake3SeqNo;
+use super::address::CryptoAddress;
 use crate::prelude::*;
-use crate::stable_hash::UnorderedAggregator;
 use blake3::Hasher;
 use ibig::UBig;
 use lazy_static::lazy_static;
 use num_traits::identities::One;
 use std::default::Default;
 
+// TODO: Consider using a Solinas prime
+/*
+From Jackson:
+    In particular we could change the prime to be a Solinas prime.
+    If we implement the algorithm for fast modular multiplication around a solinas prime then we get a big speed up.
+    So the changes would be just change the public parameter prime in the codebase, and don’t just naively multiply and reduce, but write the algorithm to take advantage of the structure inherent in Solinas primes
+    (they are prime numbers that have really low hamming weights, a sort of generalization of Mersenne primes — and so computers love these numbers)
+*/
 lazy_static! {
     static ref P: UBig = "50763434429823703141085322590076158163032399096130816327134180611270739679038131809123861970975131471260684737408234060876742190838745219274061025048845231234136148410311444604554192918702297959809128216170781389312847013812749872750274650041183009144583521632294518996531883338553737214586176414455965584933129379474747808392433032576309945590584603359054260866543918929486383805924215982747035136255123252119828736134723149397165643360162699752374292974151421555939481822911026769138419707577501643119472226283015793622652706604535623136902831581637275314074553942039263472515423713366344495524733341031029964603383".parse().unwrap();
 }
 
 /// Based on https://crypto.stackexchange.com/a/54546
 ///
-/// The idea here is to use the SequenceNumber to unambiguously identify each
+/// The idea here is to use the FieldAddress to unambiguously identify each
 /// field as within it's own database cell, and use an online order-independent
 /// aggregator of the cells to produce a final result.
 ///
 /// Within this framework a huge struct can be hashed incrementally or even in
-/// parallel as long as sequence numbers are deterministically produced to
-/// identify parts within the struct. Conveniently, the SequenceNumber::skip
+/// parallel as long as field addresses are deterministically produced to
```

Comment:
This comment is not speaking about code changes, but about hashing the struct incrementally and/or in parallel (which graph-node takes advantage of). It's saying that if you do such shenanigans, be sure to have struct.field have the same field address across runs. This comment was more salient for the previous API, which was `fieldAddress.nextChild()` instead of the current API which uses `fieldAddress.child(index)` which is much harder to screw up for a parallel implementation.

Your assertion about keeping the addresses fixed across code changes is also correct and important, just not what this comment is speaking to.